src/components/Calculator.js

```diff
@@ -1,31 +1,85 @@
+/* eslint-disable react/no-access-state-in-setstate */
+/* eslint-disable react/no-unused-state */
+/* eslint-disable react/destructuring-assignment */
 /* eslint-disable react/button-has-type */
 /* eslint-disable react/prefer-stateless-function */
 import React from 'react';
+import calculate from '../logic/calculate';
 
 export default class MyCalculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.num = '';
+    this.value = 0;
+    this.state = {
+      screen: 0, total: 0, next: 0, operation: '',
+    };
+    this.calchandler = this.calchandler.bind(this);
+  }
+
+  calchandler(e) {
+    // if (e.target.innerText === 'AC') {
+    //   // this.value = 0;
+    //   // this.num = '';
+    //   const { total, next, operation } = calculate(this.state, e.target.innerText);
+    //   // alert(total);
+    //   this.setState({
+    //     screen: 0, total, next, operation,
+    //   });
+    // } else {
+    //   // this.num += e.target.innerText;
+    //   const { total, next, operation } = calculate(this.state, e.target.innerText);
+    //   alert(total);
+    //   // this.value = total;
+    //   // alert(this.value);
+    //   this.setState({
+    //     screen: this.num, total, next, operation,
+    //   });
+    // }
+    if (e.target.innerText === 'AC') {
+      this.num = '';
+      const { total, next, operation } = calculate(this.state, e.target.innerText);
+      this.setState({
+        screen: 0, total, next, operation,
+      });
+    } else if (e.target.id === 'eqoperation') {
+      const { total, next, operation } = calculate(this.state, e.target.innerText);
+      this.setState({
+        screen: total, total, next, operation,
+      });
+      this.num = total;
+      // alert(this.state.total);
+    } else {
+      this.num += e.target.innerText;
+      this.setState({ screen: this.num, ...calculate(this.state, e.target.innerText) });
+      // alert(this.state.total);
+    }
+  }
+
   render() {
     return (
       <div className="grid-continer">
-        <input className="resultscreen" placeholder="0" />
-        <button>AC</button>
-        <button>+/-</button>
-        <button>%</button>
-        <button className="buttoperation">/</button>
-        <button>7</button>
-        <button>8</button>
-        <button>9</button>
-        <button className="buttoperation">x</button>
-        <button>4</button>
-        <button>5</button>
-        <button>6</button>
-        <button className="buttoperation">-</button>
-        <button>1</button>
-        <button>2</button>
-        <button>3</button>
-        <button className="buttoperation">+</button>
-        <button className="zeroitem">0</button>
-        <button>.</button>
-        <button className="buttoperation">=</button>
+        <input className="resultscreen" readOnly placeholder={this.state.screen} />
+        <button onClick={this.calchandler}>AC</button>
+        <button onClick={this.calchandler}>+/-</button>
```

Comment:
- [ ] Please fix the issue with this button
_See the image down below._ 🙏 😄 
![Screenshot](https://user-images.githubusercontent.com/47625407/129447855-1f6427e0-2fe9-4641-af51-f437ae72a0fc.png)

Reply:
Well addressed this!